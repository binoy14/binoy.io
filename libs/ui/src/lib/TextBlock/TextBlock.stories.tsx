import { Meta, Story } from "@storybook/react";

import Section from "../Section/Section";
import { TextBlock, TextBlockProps } from "./TextBlock";

export default {
  component: TextBlock,
  title: "TextBlock",
} as Meta;

const Template: Story<TextBlockProps> = (args) => (
  <Section type="dark">
    <TextBlock {...args} />
  </Section>
);

export const Primary = Template.bind({});
Primary.args = {
  header: "Hi, this is a header",
  subtext: "This is a sub header",
};
