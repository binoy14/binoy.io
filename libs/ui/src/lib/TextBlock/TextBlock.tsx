import { Maybe } from "@binoy14/cms-types";

export interface TextBlockProps {
  header: string;
  subtext?: Maybe<string>;
}

export function TextBlock({ header, subtext }: TextBlockProps) {
  return (
    <div className="my-5">
      <h1 className="mb-1 text-3xl">{header}</h1>
      <span>{subtext}</span>
    </div>
  );
}
