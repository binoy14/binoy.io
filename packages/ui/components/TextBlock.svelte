<script lang="ts">
  export let header: string;
  export let subtext: string | null = null;
</script>

<div class="my-5">
  <h1 class="mb-1 text-3xl">{header}</h1>
  <span>{subtext}</span>
</div>
