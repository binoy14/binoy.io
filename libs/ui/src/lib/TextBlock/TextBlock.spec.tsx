import { render } from "@testing-library/react";

import { TextBlock } from "./TextBlock";

describe("TextBlock", () => {
  it("should render successfully", () => {
    const { baseElement } = render(<TextBlock header="test" subtext="sub header" />);
    expect(baseElement).toMatchSnapshot();
  });
});
