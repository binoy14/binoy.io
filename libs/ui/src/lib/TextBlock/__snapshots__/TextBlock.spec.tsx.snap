// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextBlock should render successfully 1`] = `
<body>
  <div>
    <div
      class="my-5"
    >
      <h1
        class="mb-1 text-3xl"
      >
        test
      </h1>
      <span>
        sub header
      </span>
    </div>
  </div>
</body>
`;
